Dockerfile

```diff
@@ -1,55 +1,40 @@
-##
-#Copyright 2014 Cisco Systems Inc. All rights reserved.
-#
-#Licensed under the Apache License, Version 2.0 (the "License");
-#you may not use this file except in compliance with the License.
-#You may obtain a copy of the License at
-#http://www.apache.org/licenses/LICENSE-2.0
-#
-#Unless required by applicable law or agreed to in writing, software
-#distributed under the License is distributed on an "AS IS" BASIS,
-#WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-#See the License for the specific language governing permissions and
-#limitations under the License.
-##
-
-##
-# Container image for compiled netplugin binaries
-#
-# Run netplugin:
-# docker run --net=host <image> -host-label=<label>
-##
-
-FROM golang:1.7.6
-
-# Insert your proxy server settings if this build is running behind
-# a proxy.
-#ENV http_proxy ""
-#ENV https_proxy ""
-ARG http_proxy
-ARG https_proxy
-
-WORKDIR /go/src/github.com/contiv/netplugin/
-
-ENTRYPOINT ["netplugin"]
-CMD ["--help"]
-
-# by far, most of the compilation time is building vendor packages
-# build the vendor dependencies as a separate docker caching layer
-COPY ./vendor/ /go/src/github.com/contiv/netplugin/vendor/
-COPY ./objdb/ /go/src/github.com/contiv/netplugin/objdb/
-
-RUN GOGC=1500 go install -ldflags "-s -w" $(go list ./vendor/...)
-
-# build the netplugin binaries
-COPY ./ /go/src/github.com/contiv/netplugin/
-
-ARG BUILD_VERSION=""
-ARG NIGHTLY_RELEASE=""
-
-RUN GOPATH=/go/ \
-    BUILD_VERSION="${BUILD_VERSION}" \
-    NIGHTLY_RELEASE="${NIGHTLY_RELEASE}" \
-    make compile \
-      && cp scripts/contrib/completion/bash/netctl /etc/bash_completion.d/netctl \
-      && netplugin -version
+# builder is where netplugin got complied
+FROM golang:1.7.6 as builder
```

Comment:
Multi-stage builds are only supported in 17.05 or >

Will this change affect anything relying on this var?  https://github.com/contiv/netplugin/blob/0308c0f45158412968e0e899430c3736481338e8/Makefile#L8

Reply:
This is used for dockerhub auto build(and they have newer enough docker), in our local build though, i think we should upgrade if its not supported (and/or provide a fallback if it's a customer ask)

Reply:
if you run centos and maybe ubuntu, you get the older version, you have to do extra work to get the newer stuff, I'm not personally opposed to requiring newer docker considering we're not supporting older docker runtime deployments . .

Reply:
it doesn't impact where running the container, it only impacts where build the container